compiler/compiler/src/test.rs

```diff
@@ -205,19 +221,49 @@ fn run_test(test: Test, handler: &Handler, err_buf: &BufferEmitter) -> Result<Va
     let mut output_items = Vec::with_capacity(inputs.len());
 
     if inputs.is_empty() {
-        handler.extend_if_error(compile_and_process(&mut parsed))?;
         output_items.push(OutputItem {
             initial_input_ast: "no input".to_string(),
         });
     } else {
+        // Parse one or more input files to execute the program with.
         for input in inputs {
             let mut parsed = parsed.clone();
             handler.extend_if_error(parsed.parse_input(input))?;
-            handler.extend_if_error(compile_and_process(&mut parsed))?;
             let initial_input_ast = hash_file("/tmp/output/initial_input_ast.json");
 
             output_items.push(OutputItem { initial_input_ast });
         }
+    };
+
+    // Compile the program to bytecode.
+    let program_name = format!("{}.aleo", parsed.program_name);
+    let bytecode = handler.extend_if_error(compile_and_process(&mut parsed, handler))?;
+
```

Comment:
We should consider taking a hash of the bytecode and storing it in the output.

Reply:
Agreed. New issue #1980